src/grid.tsx

```diff
@@ -0,0 +1,209 @@
+        }
+        if (typeof rows.maxSize !== 'undefined') {
+            autoRepeatVars[styles.vars.rowMaxSize] = normalizeMaxSize(rows.maxSize);
+        }
+    }
+
+    if (columns && typeof columns !== 'number') {
+        if (typeof columns.minSize !== 'undefined') {
+            autoRepeatVars[styles.vars.columnMinSize] = normalizeMinSize(columns.minSize);
+        }
+        if (typeof columns.maxSize !== 'undefined') {
+            autoRepeatVars[styles.vars.columnMaxSize] = normalizeMaxSize(columns.maxSize);
+        }
+    }
+    return autoRepeatVars;
+};
+
+type AutoFillConfig = {
+    minSize?: string | number;
+    /**
+     * default: 1fr
+     */
+    maxSize?: string | number;
+};
+
+type GridProps = {
+    rows?: RowsColumns | AutoFillConfig;
+    columns?: RowsColumns | AutoFillConfig;
+    gap?: Gap | [Gap, Gap];
+    flow?: 'column' | 'row' | 'column dense' | 'row dense';
+    justifyItems?: 'start' | 'end' | 'center' | 'stretch';
+    alignItems?: 'start' | 'end' | 'center' | 'stretch' | 'baseline';
+    justifyContent?:
+        | 'start'
+        | 'end'
+        | 'center'
+        | 'stretch'
+        | 'space-around'
+        | 'space-between'
+        | 'space-evenly';
+    alignContent?: 'start' | 'end' | 'center' | 'stretch' | 'space-around' | 'space-between' | 'space-evenly';
+    width?: string | number;
+    height?: string | number;
+    minWidth?: string | number;
+    minHeight?: string | number;
+    children: React.ReactNode;
+    as?: React.ComponentType<any> | string;
+    /** "data-" prefix is automatically added. For example, use "testid" instead of "data-testid" */
+    dataAttributes?: DataAttributes;
+};
+
+export const Grid = React.forwardRef<HTMLDivElement, GridProps>(
+    (
+        {
+            rows,
+            columns,
+            gap = 0,
+            flow,
+            justifyItems,
+            alignItems,
+            justifyContent,
+            alignContent,
+            width,
+            height,
+            minWidth,
+            minHeight,
+            children,
+            as = 'div',
+            dataAttributes,
+        },
+        ref
+    ) => {
+        const [columnGap, rowGap] = Array.isArray(gap) ? gap : [gap, gap];
+
+        return React.createElement(
+            as,
+            {
+                ref,
+                className: classNames(
+                    styles.grid,
+                    columns
+                        ? typeof columns === 'number'
+                            ? styles.gridTemplateColumns[columns]
+                            : styles.gridTemplateColumnsAutoRepeat
+                        : '',
+                    rows
+                        ? typeof rows === 'number'
+                            ? styles.gridTemplateRows[rows]
+                            : styles.gridTemplateRowsAutoRepeat
+                        : '',
+                    flow ? styles.gridAutoFlow[flow] : '',
```

Comment:
There is no need for the undefined check in here. If `flow` is undefined, the resulting value will be undefined, and `classnames` works fine when that happens.

Reply:
I need the check, otherwise TS will error:
![image](https://github.com/Telefonica/mistica-web/assets/1849576/41762725-0093-4f65-bf9c-d6f211b6830b)